Adjusts the description of regression and classification algorithms (#2338)
elastic/stack-docs: docs/en/stack/ml/df-analytics/ml-dfa-classification.asciidoc

//tag::classification-algorithms[]
{classanalysis-cap} uses an ensemble algorithm that is a type of boosting called 

> extreme gradient boost (XGboost) which combines multiple weak models into a 

Reviewer: ```suggestion
extreme gradient boosting (XGBoost), which combines multiple weak models into a 
```

Reviewer: It should say that our ensemble algorithm is **similar to** XGBoost. As the development of our code progressed, we diverged in quite a number of details and implementation decisions from the "original" XGBoost.

composite one. It uses decision trees to learn to predict the probability that a 
data point belongs to a certain class. XGboost trains a sequence of decision 

> trees and avery decision tree learns from the mistakes of the previous one. 

Reviewer: ```suggestion
trees and every decision tree learns from the mistakes of the forest so far. 
```

Every tree is an iteration of the last one, hence it improves the decision made 

> by the previous tree. The classification algorithm is optimized for a loss 

Reviewer: Thus, in each iteration, we add trees that improve the decision quality of the combined decision forest.

Reviewer: The classification algorithm **optimizes for** ...

> function called cross-entropy loss by default.

Reviewer: ```suggestion
function called cross-entropy loss.
```
